Build a regression model.

In [11]:
import pandas as pd
import sqlite3
from sqlite3 import Error
import statsmodels.api as sm

In [ ]:
def create_connection(path):
    connection = None
    try:
        connection = sqlite3.connect(path)
        print("Connection to SQLite DB successful")
    except Error as e:
        print(f"The error '{e}' occurred")

    return connection

In [ ]:
connection = create_connection("sm_app.sqlite")

In [28]:
df = pd.read_sql('SELECT * FROM pois', connection)

In [29]:
df = df.drop(['index'], axis=1)
df = df.drop(['station name', 'name', 'address', 'category'], axis=1)
df = df.dropna()

In [35]:
#X = df[df.columns[~df.columns.isin(['free_bikes'])]]
X = df['rating']
y = df['free_bikes']

X = sm.add_constant(X)
lin_reg = sm.OLS(y,X)

Provide model output and an interpretation of the results. 

In [36]:
model = lin_reg.fit()
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:             free_bikes   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.004
Method:                 Least Squares   F-statistic:                   0.09665
Date:                Sun, 30 Jul 2023   Prob (F-statistic):              0.756
Time:                        11:53:42   Log-Likelihood:                -464.11
No. Observations:                 223   AIC:                             932.2
Df Residuals:                     221   BIC:                             939.0
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.8824      0.765      1.154      0.2

Notes
- The R-squared value is 0.000 which means the model is fit
- However the p-value is 0.756 > 0.05 so the rating doesn't have significant impact to the number of bikes

# Stretch

How can you turn the regression model into a classification model?